pkg/nri/config.go

```diff
@@ -17,42 +17,66 @@
 package nri
 
 import (
+	"time"
+
 	nri "github.com/containerd/nri/pkg/adaptation"
 )
 
 // Config data for NRI.
 type Config struct {
 	// Disable this NRI plugin and containerd NRI functionality altogether.
 	Disable bool `toml:"disable" json:"disable"`
-	// ConfigPath is the path to the NRI configuration file to use.
-	ConfigPath string `toml:"config_file" json:"configFile"`
 	// SocketPath is the path to the NRI socket to create for NRI plugins to connect to.
 	SocketPath string `toml:"socket_path" json:"socketPath"`
 	// PluginPath is the path to search for NRI plugins to launch on startup.
 	PluginPath string `toml:"plugin_path" json:"pluginPath"`
+	// PluginConfigPath is the path to search for plugin-specific configuration.
+	PluginConfigPath string `toml:"plugin_config_path" json:"pluginConfigPath"`
+	// PluginRegistrationTimeout is the timeout for plugin registration.
+	PluginRegistrationTimeout time.Duration `toml:"plugin_registration_timeout" json:"pluginRegistrationTimeout"`
+	// PluginRequestTimeout is the timeout for a plugin to handle a request.
+	PluginRequestTimeout time.Duration `toml:"plugin_request_timeout" json:"pluginRequestTimeout"`
+	// DisableConnections disables connections from externally launched plugins.
+	DisableConnections bool `toml:"disable_connections" json:"disableConnections"`
 }
 
 // DefaultConfig returns the default configuration.
 func DefaultConfig() *Config {
 	return &Config{
-		Disable:    true,
-		ConfigPath: nri.DefaultConfigPath,
-		SocketPath: nri.DefaultSocketPath,
-		PluginPath: nri.DefaultPluginPath,
+		Disable:          true,
+		SocketPath:       nri.DefaultSocketPath,
```

Comment:
is odd to be retrieving containerd config.toml defaults values from constants in a vendor'd package.. should these config defaults be specified here in the plugin?

Reply:
We considered the (well-known default) socket path to be part of the (runtime-agnostic) NRI API, so we defined it on the NRI side of the fence and made the configurable NRI socket path default to that in both NRI-enabled runtimes.

I think the alternative would have been (and still is) to instead co-locate the NRI socket with the runtime one in the runtime-specific directory and then take on the NRI side the `cri-tools`ish approach of probing sockets in known runtime-specific locations (unless specifically told otherwise) until one is found that accepts connections. However, if we jump to the NRI side of the fence, then this does not feel right. ATM NRI is runtime agnostic and is unaware of any runtime-specific details. It is the runtimes that already know about NRI.

No too strong feelings/opinion leaning in either direction. Just trying to provide some background how we ended up opting for the current choice...